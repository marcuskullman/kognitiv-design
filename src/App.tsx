import { FC, useReducer, ChangeEvent } from "react"
import Form from "./components/form"
import {
  useTTest,
  usePermutationTest,
  useAbTest,
  useSampleSizeCalculator,
} from "./hooks"

const App: FC = () => {
  const mockdata1 = require("./test/mockdata-1.json")
  const mockdata2 = require("./test/mockdata-2.json")

  const unpairedTTest = useTTest({ dataset: mockdata1, type: "unpaired" })
  console.log("Unpaired T-test:", unpairedTTest)

  const pairedTTest = useTTest({ dataset: mockdata1, type: "paired" })
  console.log("Paired T-test:", pairedTTest)

  const { samples, permutationTest } = usePermutationTest({
    dataset: mockdata2,
  })
  console.log("Samples:", samples)
  console.log("Permutation test:", permutationTest)

  const abTest = useAbTest(samples)
  console.log("A/B-test", abTest)

  // Nedan Ã¤r relaterat till Provstorlekskalkylatorn

  const [{ MDE, CR }, setState]: IState = useReducer(
    (state: IDefaultState, action: IDefaultStateDraft) => ({
      ...state,
      ...action,
    }),
    {
      MDE: samples.testStatistic,
      CR: (100 * (samples.a.conversionRate + samples.b.conversionRate)) / 2,
    }
  )

  const props = {
    MDE,
    CR,
    handleChange: ({
      target: { name, value },
    }: ChangeEvent<HTMLInputElement>): void =>
      setState({ [name]: Number(value) }),
  }

  const sampleSize = useSampleSizeCalculator({
    ...samples,
    mean: abTest.mean,
    MDE,
    CR,
  })

  console.log("Sample size:", sampleSize)

  return <Form {...props} />
}

export default App
