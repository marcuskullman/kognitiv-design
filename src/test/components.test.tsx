import { render, screen } from "@testing-library/react"
import App from "../App"

describe("App", () => {
  test("Renders", () => {
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()

    const form = screen.getByTestId("form")
    expect(form).toBeInTheDocument()
  })
})
