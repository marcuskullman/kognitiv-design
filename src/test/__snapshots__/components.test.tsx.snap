// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders 1`] = `
<div>
  <form
    data-testid="form"
  >
    <label>
      MDE:
    </label>
    <input
      name="MDE"
      type="number"
      value="66.66666666666667"
    />
    %
    <br />
    <br />
    <label>
      CR
    </label>
    <input
      name="CR"
      type="number"
      value="40"
    />
    %
  </form>
</div>
`;
